java/parser/Parser.java

```diff
@@ -897,25 +899,42 @@ public Object visitValue(GraqlParser.ValueContext ctx) {
             return getDateTime(ctx.DATETIME_());
 
         } else {
-            throw new IllegalArgumentException("Unrecognised Literal token: " + ctx.getText());
+            throw GraqlException.of(PARSING_ERROR_LITERAL_TOKEN.message(ctx.getText()));
         }
     }
 
     private String getString(TerminalNode string) {
-        // Remove surrounding quotes
-        return unquoteString(string);
+        if (string.getText().length() >= 2) {
+            final GraqlToken.Char literalFirstSymbol = GraqlToken.Char.of(string.getText().substring(0,1));
+            final GraqlToken.Char literalLastSymbol = GraqlToken.Char.of(string.getText().substring(string.getText().length() - 1));
+            if (literalFirstSymbol != null && literalFirstSymbol.equals(GraqlToken.Char.QUOTE) &&
+                literalLastSymbol != null && literalLastSymbol.equals(GraqlToken.Char.QUOTE)) {
+                // Remove surrounding quotes
+                return unquoteString(string);
+            }
+        }
+        throw GraqlException.of(PARSING_ERROR_QUOTEDSTRING_TOKEN.message(string.getText()));
```

Comment:
thrown in case the length of the text is less than 2 OR it does not end with a quote symbol. Writing it in else twice will look ugly, so I just throw it in the end